analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/ComplianceAnalyzerPlugin.java

```diff
@@ -0,0 +1,231 @@
+package eu.fasten.analyzer.complianceanalyzer;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.collect.Lists;
+import eu.fasten.core.plugins.DBConnector;
+import eu.fasten.core.plugins.KafkaPlugin;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.Configuration;
+import io.kubernetes.client.openapi.apis.BatchV1Api;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1Job;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.util.Config;
+import io.kubernetes.client.util.KubeConfig;
+import io.kubernetes.client.util.Yaml;
+import org.jooq.DSLContext;
+import org.json.JSONObject;
+import org.pf4j.Extension;
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ * Plugin which runs qmstr command line tool to detect
+ * license compatibility and compliance.
+ */
+public class ComplianceAnalyzerPlugin extends Plugin {
+
+    /**
+     * Kubernetes namespace where our demo objects are going to be deployed in.
+     */
+    protected static final String K8S_NAMESPACE = "default";
+
+    public ComplianceAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class CompliancePluginExtension implements KafkaPlugin {
+
+        private final Logger logger = LoggerFactory.getLogger(CompliancePluginExtension.class.getName());
+
+        protected String consumerTopic = "fasten.RepoCloner.out";
+        protected Throwable pluginError = null;
+        protected String repoUrl;
+
+        /**
+         * Path to the Kubernetes cluster credentials file
+         */
+        protected final String clusterCredentialsFilePath;
+
+        public CompliancePluginExtension(String clusterCredentialsFilePath) {
+            this.clusterCredentialsFilePath = clusterCredentialsFilePath;
```

Comment:
Pf4j needs a default constructor to instantiate an Extension. It would have been better to pass the access token to GoogleCredentials Builder in an environmental variable. I took a quick look at the GoogleCredentials class and as far as I understand you don't necessarily need a credentials file, but just an access token. Use environmental variable  for this, because otherwise we will have to introduce a new parameter on the Fasten server to pass path to file / token to the Compliance plugin, which will be plugin-specific (not preferable). Even if access token is not enough, then pass the path to the credentials file, but still thorough env variables.

Reply:
Transformed in an ENV var in bb8018d1.